### ISSUE TITLE
Cloudflare Hyperdrive

### ISSUE THREAD

**Issue by MehrdadKhnzd** [withastro/adapters#119]
### Astro Info

```block
Astro                    v4.0.8
Node                     v20.10.0
System                   Linux (x64)
Package Manager          bun
Output                   server
Adapter                  @astrojs/cloudflare
Integrations             unocss
                         @astrojs/solid-js
                         @swup/astro
```

### Describe the Bug

Cloudflare Hyperdrive doesn't work either in production using Cloudflare Workers or locally (via the provided `localConnectionString`). When I try to read it in my endpoints as `locals.runtime.env.HD.connectionString`, `HD` is `undefined` and it throws an error.

`wrangler.toml`:
```toml
[[hyperdrive]]
binding = "HD"
id = "XXX" # Hyperdrive ID
```

`astro.config.mjs`:
```mjs
{
  ...
  adapter: cloudflare({
    mode: 'advanced',
    functionPerRoute: false,
    runtime: {
      mode: 'local',
      type: 'pages',
      bindings: {
        'HD': {
          type: 'hyperdrive',
          id: 'XXX', # Hyperdrive ID
          localConnectionString: import.meta.env.DB_CONNECTION,
        }
      }
    }
  }),
  ...
}
```

`env.d.ts`:
```ts
/// <reference types="astro/client" />

type Hyperdrive = import("@cloudflare/workers-types").Hyperdrive
type ENV = {
  HD: Hyperdrive
}

type Runtime = import("@astrojs/cloudflare").AdvancedRuntime<ENV>
declare namespace App {
  interface Locals extends Runtime {
    ...
  }
}
```

### What's the expected result?

It should work, at least when deployed.

### Link to Minimal Reproducible Example

N/A

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by alexanderniebuhr** [comment]
Hyperdrive is not added yet, but if Miniflare supports it, we can add it

**Comment by alexanderniebuhr** [comment]
We will switch our implementation to a Cloudflare exposed utility, which will have all bindings included (see #153)
